http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java

```diff
@@ -81,7 +82,15 @@ boolean acquireStream(Promise<Channel> promise) {
     private void acquireClaimedStream(Promise<Channel> promise) {
         doInEventLoop(connection.eventLoop(), () -> {
             if (state != RecordState.OPEN) {
-                String message = "Connection received GOAWAY or was closed while acquiring new stream.";
+                String message;
+                // GOAWAY
+                if (state == RecordState.CLOSED_TO_NEW) {
+                    message = String.format("Connection %s received GOAWAY with Last Stream ID %d. Unable to open new "
+                                            + "streams on this connection.", connection, lastStreamId);
+                } else {
+                    message = String.format("Connection %s was closed while acquiring new stream.", connection);
+                }
+                log.warn(() -> message);
```

Comment:
Same here. Can we add the message in the exception instead?

Reply:
I would rather keep this here. This is the specific case we care about right, trying to create a stream on one that has received a `GOAWAY`. The message is added to the exception as well